### ISSUE TITLE
Auto align option with C-c C-space

### ISSUE THREAD

**Issue by BrachystochroneSD** [kopoli/robot-mode#9]
Currently, the alignment of function is done it two times. 
First to put your keyword, arguments with `C-c C-space`, etc... and then you hit `C-c C-a` to align the current defun.

Why not making the `C-c C-space` automatically put the cursor where it should be, aligned with the previous line ?
If people don't whant to align by default, a customizable variable could be set for auto-align, something like `robot-auto-align-arguments`

A workaround is to use `C-c C-a` instead of `C-c C-space` but the cursor doesn't go where it should, and it is kind of inefficient to realign the whole block instead of just align the current line.

**Comment by kopoli** [comment]
Sorry, I have missed this completely.

Often I just use the `C-c C-a`, as you describe, to clean up the whole or only parts of a defun. Mostly for making them more aesthetic.

It would be probably good to test how it would behave if `C-c C-space` would automatically sniff the alignment from the previous row. Immediately the following potential problems come into mind:

1. Will the alignment between some cells become longer and longer whenever more steps are added to a defun ?
2. Will it be problematic if the alignment becomes out of sync if steps are added or removed in the middle of a defun ? 
3. If one does not want to align the following rows according to the previous, would there be some rules (e.g. two empty rows would disable, or maybe a key press to toggle the `robot-auto-align-arguments`) ?

Also the official documentation does not exhibit this kind of an alignment:
https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#toc-entry-342

**Comment by BrachystochroneSD** [comment]
Indeed, since it is not official in the style guide. I guess it's not really worth spending time on it
